autorest/authorization.go

```diff
@@ -115,19 +115,23 @@ func (ba *BearerAuthorizer) withBearerAuthorization() PrepareDecorator {
 func (ba *BearerAuthorizer) WithAuthorization() PrepareDecorator {
 	return func(p Preparer) Preparer {
 		return PreparerFunc(func(r *http.Request) (*http.Request, error) {
-			refresher, ok := ba.tokenProvider.(adal.Refresher)
-			if ok {
-				err := refresher.EnsureFresh()
-				if err != nil {
-					var resp *http.Response
-					if tokError, ok := err.(adal.TokenRefreshError); ok {
-						resp = tokError.Response()
+			r, err := p.Prepare(r)
+			if err == nil {
+				refresher, ok := ba.tokenProvider.(adal.Refresher)
+				if ok {
+					err := refresher.EnsureFresh()
+					if err != nil {
+						var resp *http.Response
+						if tokError, ok := err.(adal.TokenRefreshError); ok {
+							resp = tokError.Response()
+						}
+						return r, NewErrorWithError(err, "azure.BearerAuthorizer", "WithAuthorization", resp,
+							"Failed to refresh the Token for request to %s", r.URL)
 					}
-					return r, NewErrorWithError(err, "azure.BearerAuthorizer", "WithAuthorization", resp,
-						"Failed to refresh the Token for request to %s", r.URL)
 				}
+				return Prepare(r, WithHeader(headerAuthorization, fmt.Sprintf("Bearer %s", ba.tokenProvider.OAuthToken())))
```

Comment:
>WithHeader(headerAuthorization, fmt.Sprintf("Bearer %s", ba.tokenProvider.OAuthToken()) [](start = 22, length = 87)

why not still use the `withBearerAuthorization` from above ?

Reply:
Ooops I actually meant to delete that function, will fix.